core/src/main/java/org/apache/calcite/util/format/postgresql/DateStringFormatPattern.java

```diff
@@ -40,19 +41,33 @@ public class DateStringFormatPattern<T> extends StringFormatPattern {
    *           that has a string representation
    */
   private interface DateStringConverter<T> {
+    ChronoUnitEnum getChronoUnit();
+
     T getValueFromDateTime(ZonedDateTime dateTime);
 
+    T[] values();
+
     String getDisplayName(T value, TextStyle textStyle, boolean haveFillMode, Locale locale);
+
+    int getValue(T value);
```

Comment:
These functions could use some documentation.
What is the integer result?

Reply:
Added more Javadoc comments, including here. Hopefully this is explained better now.